commerce/domain/checkout/order/createOrder/classes/CreateOrderSampleUnitTest.cls

```diff
@@ -0,0 +1,147 @@
+ * the test is isolated from the dependencies on other services that default implementation might have.
+ */
+@isTest
+public class CreateOrderSampleUnitTest {
+    
+    @isTest
+    public static void shouldRoundUpCostOfProduct() {
+        CartExtension.Cart cart = arrangeCart(new Decimal[]{1.5});
+        CreateOrderSample createOrder = new CreateOrderSampleMock();
+        // Act
+        Test.startTest();
+        CartExtension.CreateOrderResponse response = createOrder.createOrder(new CartExtension.CreateOrderRequest(cart, '123'));
+        Test.stopTest();
+        // Assert
+        Order order = response.getOrderGraph().getOrder();
+        Assert.areEqual('Total rounded up cost of products: 2.0', order.Description);
+    }
+
+    @isTest
+    public static void shouldRoundUpCostOfSeveralProducts() {
+        CartExtension.Cart cart = arrangeCart(new Decimal[]{1.5, 2.6});
+        CreateOrderSample createOrder = new CreateOrderSampleMock();
+        // Act
+        Test.startTest();
+        CartExtension.CreateOrderResponse response = createOrder.createOrder(new CartExtension.CreateOrderRequest(cart, '123'));
+        Test.stopTest();
+        // Assert
+        Order order = response.getOrderGraph().getOrder();
+        Assert.areEqual('Total rounded up cost of products: ' + (2.0 + 3.0), order.Description);
+    }
+    
+    /*
+     * Mock of CreateOrderSample class. It provides simplified implementation of cart to order mapping. 
+     */    
+    public class CreateOrderSampleMock extends CreateOrderSample {
+        
+        public override CartExtension.CreateOrderResponse callDefault(CartExtension.CreateOrderRequest request) {
+            CartExtension.Cart cart = request.getCart();
+            CartExtension.OrderGraph orderGraph = new CartExtension.OrderGraph();
+            Order order = orderGraph.getOrder();
+            System.assert(order.Id != null);
+    
+            order.AccountId = cart.getAccountId();
+            order.OwnerId = cart.getOwnerId();
+            order.SalesStoreId = cart.getWebStoreId();
+            // cart does not support PoNumber yet
+            //order.PoNumber = cart.getPoNumber();
+            Address billingAddress = cart.getBillingAddress();
+            if (billingAddress != null) {
+                order.BillingStreet = billingAddress.Street;
+                order.BillingCity = billingAddress.City;
+                order.BillingState = billingAddress.State;
+                order.BillingPostalCode = billingAddress.PostalCode;
+                order.BillingCountry = billingAddress.Country;
+                order.BillingLatitude = billingAddress.Latitude;
+                order.BillingLongitude = billingAddress.Longitude;
+                order.BillingEmailAddress = cart.getGuestEmailAddress();
+                order.BillingPhoneNumber = cart.getGuestPhoneNumber();
+                order.TaxLocaleType = cart.getTaxType().name();
+                order.EffectiveDate = Date.today();
```

Comment:
@yan-huang-sf this field is missing in the OrderSummary but it's required for Order. Seems we'll have to deal with it somehow.

Reply:
Not sure how HSOI will set this field when they create order